sshkey.c

```diff
@@ -3415,8 +3421,16 @@ sshkey_parse_private2(struct sshbuf *blob, int type, const char *passphrase,
 	/* check preamble */
 	cp = sshbuf_ptr(blob);
 	encoded_len = sshbuf_len(blob);
+	
+#ifdef SUPPORT_CRLF
+	if ((encoded_len < (MARK_BEGIN_LEN + MARK_END_LEN) ||
+	    memcmp(cp, MARK_BEGIN, MARK_BEGIN_LEN) != 0) &&
+	    (encoded_len < (MARK_BEGIN_LEN_CRLF + MARK_END_LEN_CRLF) ||
+	    memcmp(cp, MARK_BEGIN_CRLF, MARK_BEGIN_LEN_CRLF) != 0)) {
+#else  /* !SUPPORT_CRLF */
 	if (encoded_len < (MARK_BEGIN_LEN + MARK_END_LEN) ||
```

Comment:
Missed this in earlier review. So you want to support only one of the format at any time? Can we support both formats on Windows?

Reply:
this change support both formats on windows